cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -351,27 +350,72 @@ public TimeSeries getTimeseries(String page, int pageSize, String names, String
-            );
+            VersionType finalDateVersionType = getVersionType(dsl, names, office, versionDate != null);
+            if (!includeEntryDate) {
+                return new TimeSeries(recordCursor, recordPageSize, tsMetadata.getValue("TOTAL",
+                        Integer.class), tsMetadata.getValue("NAME", String.class),
+                        tsMetadata.getValue("office_id", String.class),
+                        beginTime, endTime, tsMetadata.getValue("units", String.class),
+                        Duration.ofMinutes(tsMetadata.get("interval") == null ? 0 :
+                                tsMetadata.getValue("interval", Long.class)),
+                        verticalDatumInfo,
+                        tsMetadata.getValue(AV_CWMS_TS_ID2.INTERVAL_UTC_OFFSET).longValue(),
+                        tsMetadata.getValue(tzName),
+                        versionDate, finalDateVersionType
+                );
+            } else {
+                return new TimeSeriesWithDate(recordCursor, recordPageSize, tsMetadata.getValue("TOTAL",
+                        Integer.class), tsMetadata.getValue("NAME", String.class),
+                        tsMetadata.getValue("office_id", String.class),
+                        beginTime, endTime, tsMetadata.getValue("units", String.class),
+                        Duration.ofMinutes(tsMetadata.get("interval") == null ? 0 :
+                                tsMetadata.getValue("interval", Long.class)),
+                        verticalDatumInfo,
+                        tsMetadata.getValue(AV_CWMS_TS_ID2.INTERVAL_UTC_OFFSET).longValue(),
+                        tsMetadata.getValue(tzName),
+                        versionDate, finalDateVersionType
+                );
+            }
         });
 
+        if (includeEntryDate) {
+            timeseries = new TimeSeriesWithDate(timeseries);
+        }
+
+
         Field<Timestamp> dataEntryDate = field("DATA_ENTRY_DATE", Timestamp.class).as("data_entry_date");
+        Condition whereCond = dateTimeCol
+                .greaterOrEqual(CWMS_UTIL_PACKAGE.call_TO_TIMESTAMP__2(
+                        DSL.nvl(DSL.val(tsCursor == null ? null :
+                                        tsCursor.toInstant().toEpochMilli()),
+                                DSL.val(beginTime.toInstant().toEpochMilli()))))
+                .and(dateTimeCol
+                        .lessOrEqual(CWMS_UTIL_PACKAGE.call_TO_TIMESTAMP__2(
+                                DSL.val(endTime.toInstant().toEpochMilli())))
+                        .and(AV_TSV_DQU.AV_TSV_DQU.CWMS_TS_ID.equalIgnoreCase(names))
+                        .and(AV_TSV_DQU.AV_TSV_DQU.OFFICE_ID.eq(office))
+                        .and(AV_TSV_DQU.AV_TSV_DQU.UNIT_ID.equalIgnoreCase(unit)));
 
+        TimeSeries retVal = null;
         if (pageSize != 0) {
+            if (versionDate != null) {
+                whereCond = whereCond.and(AV_TSV_DQU.AV_TSV_DQU.VERSION_DATE.eq(versionDate == null ? null :
```

Comment:
Does this logic handle max version? or will AV_TSV_DQU always return every version? or only the specifically requested version?

Reply:
Updated logic to handle max version in line with previous implementation. If the version date is not specified, the max version will be set to true and the most recent version date will be used for retrieval. This does prevent retrieval of multiple versions in one call, so that could be a problem if that is a desired feature.